### PR TITLE
Create dependabot.yml

### PR DESCRIPTION
**Pull request by inanc-can**
Dependabot is a valuable tool designed to assist in maintaining up-to-date dependencies. It achieves this by automatically generating pull requests whenever a new version of a dependency becomes available, ensuring a seamless integration without any disruptive changes.

In this pull request, the inclusion of the dependabot.yml file is crucial. This file is responsible for configuring version updates specifically for GitHub Actions, and it directs the updates to the .github/workflows directory.

To fully utilize Dependabot, a maintainer may need to enable it through the following steps: navigate to the settings section, then access the Code security and analysis options. Within this section, there will be three buttons that need to be enabled to activate Dependabot's functionalities.

### CODE REVIEW

File: .github/workflows/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
```
Comment (hkaiser) [inline .github/workflows/dependabot.yml:1]: Please add License, copyright, and SPDX entry to this file:
```
# Copyright (c) 2023 <your name>
#
# SPDX-License-Identifier: BSL-1.0
# Distributed under the Boost Software License, Version 1.0. (See accompanying
# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
```

**Comment by hkaiser** [comment]
Congratulations, your first PR to the HPX repository has just been merged! As a 'thank you' we offer a free STE||AR-Group t-shirt to all of our first-time contributors. If you are interested in receiving one, please get back to me directly so we can set up the delivery.

**Comment by hkaiser** [comment]
@inanc-can we're now seeing errors like: https://github.com/STEllAR-GROUP/hpx/actions/runs/5122998883. Does your PR miss something?

**Comment by inanc-can** [comment]
Corrected the mistake and created a new pull request  => https://github.com/STEllAR-GROUP/hpx/pull/6275

**Comment by hkaiser** [comment]
> Hello, I live in Germany and you can find my address here. ... deleted address for privacy reasons Please get back to me if any further detail is necessary.

What t-shirt size would you prefer? Also would you like to have a white or a black one?

**Comment by hkaiser** [comment]
Sure, consider it being on its way. Thanks!

**Pull request merged**